### ISSUE TITLE
Handling constant subsequence

### ISSUE THREAD

**Issue by oentaryorj** [target/matrixprofile-ts#11]
In some occasion, a time series subsequence can be flat/constant for a while, which may lead to a zero standard deviation for that subsequence. However, the the current codes does not seem to handle this case yet. Is there a way to fix this?

**Comment by aouyang1** [comment]
@oentaryorj Something that's worked for me was to add some very small noise to the signal. This has the added benefit that the flat regions can be detected later on as different segments of the time series.

**Comment by oentaryorj** [comment]
@vanbenschoten: Sorry for the late response - just back from vacation :) I would be more inclined toward option 2, although we could probably put an option to ignore/skip the flat segments.

@aouyang1: Thanks for the suggestion. Do you need to modify the current codes to make this work? Any other numerical issues encountered?

**Comment by ameya98** [comment]
If you read the original [SAX](https://cs.gmu.edu/~jessica/SAX_DAMI_preprint.pdf) paper (SAX was invented by the same people who invented the Matrix Profile) you will see that they handle subsequences with very low standard deviation (that includes the constant subsequence, as that has standard deviation zero) by not dividing by the standard deviation when normalizing. (The threshold for the standard deviation is a parameter here.) 
I think incorporating that into the matrix profile makes the most sense.

**Comment by aouyang1** [comment]
Agreed. Found this from stack overflow as well https://github.com/scikit-learn/scikit-learn/blob/7389dbac82d362f296dc2746f10e43ffa1615660/sklearn/preprocessing/data.py#L70 treating the standard deviation as 1 keeping the existing subsequence values and not normalizing. 

Since we're trying to compare to subsequences together such that one subsequence's spectral power does not bias over another, would it make sense to divide the signal such that we satisfy parseval's theorem?

**Comment by oentaryorj** [comment]
Thanks for the inputs. So I guess "ignoring" the standard deviation means option 2 in @vanbenschoten's reply (i.e., treating constant subsequence as "trivial" motif)?

**Comment by ameya98** [comment]
I think performing the mean subtraction but not the division by the standard deviation in the distance profile computation should suffice as well?
